doxygen: shell

```
$ make doc 2>&1 |grep "shell.h"
Preprocessing /home/c/git/RIOT-OS/RIOT/sys/include/shell.h...
Parsing file /home/c/git/RIOT-OS/RIOT/sys/include/shell.h...
/home/c/git/RIOT-OS/RIOT/sys/include/shell.h:33: warning: Compound shell_t is not documented.
/home/c/git/RIOT-OS/RIOT/sys/include/shell.h:27: warning: Compound shell_command_t is not documented.
Generating code for file shell.h...
/home/c/git/RIOT-OS/RIOT/sys/include/shell.h:28: warning: Member name (variable) of class shell_command_t is not documented.
/home/c/git/RIOT-OS/RIOT/sys/include/shell.h:29: warning: Member desc (variable) of class shell_command_t is not documented.
/home/c/git/RIOT-OS/RIOT/sys/include/shell.h:30: warning: Member handler (variable) of class shell_command_t is not documented.
/home/c/git/RIOT-OS/RIOT/sys/include/shell.h:34: warning: Member command_list (variable) of class shell_t is not documented.
/home/c/git/RIOT-OS/RIOT/sys/include/shell.h:35: warning: Member shell_buffer_size (variable) of class shell_t is not documented.
/home/c/git/RIOT-OS/RIOT/sys/include/shell.h:36: warning: Member readchar)(void) (variable) of class shell_t is not documented.
/home/c/git/RIOT-OS/RIOT/sys/include/shell.h:37: warning: Member put_char)(int) (variable) of class shell_t is not documented.
```

--- a/sys/include/shell.h
+++ b/sys/include/shell.h
@@ -52,10 +52,10 @@ typedef struct shell_command_t {
  *                  and shell_run() to run the REPL session.
  */
 typedef struct shell_t {
-    const shell_command_t *command_list;
-    uint16_t shell_buffer_size;
-    int (*readchar)(void);
-    void (*put_char)(int);
+    const shell_command_t *command_list; /**< The commandlist supplied to shell_init(). */
+    uint16_t shell_buffer_size;          /**< The maximum line length supplied to shell_init(). */
+    int (*readchar)(void);               /**< The read function supplied to shell_init(). */
+    void (*put_char)(int);               /**< The write function supplied to shell_init(). */
 } shell_t;
 
 /**

